solc_json_parser/parser.py

```diff
@@ -1043,7 +1043,10 @@ def _process_other_literal_node(literal_node, literals, only_value):
                 elif literal.sub_type.startswith("int"):
                     if only_value:
                         if literal.str_value.startswith('0x'):
-                            literals['number'].add(int(literal.str_value, 16))
+                            if 40 <= len(literal.str_value) <= 42:
```

Comment:
I think this is a hack better handled on the API caller side. 

Another minor issue here is that inconsistent return results depending on whether `only_value` is used, 

```
In [7]: parser.get_literals('Delta').get('number')
Out[7]:
{Literal(token_type='number', sub_type='int_const 1', str_value='1', hex_value='31'),
 Literal(token_type='number', sub_type='int_const 10', str_value='10', hex_value='3130'),
 Literal(token_type='number', sub_type='int_const 1000', str_value='1000', hex_value='31303030'),
 Literal(token_type='number', sub_type='int_const 18', str_value='18', hex_value='3138'),
 Literal(token_type='number', sub_type='int_const 6882...(40 digits omitted)...9008', str_value='0x788c45dd60ae4dbe5055b5ac02384d5dc84677b0', hex_value='307837383863343564643630616534646265353035356235616330323338346435646338343637376230'),
 Literal(token_type='number', sub_type='int_const 6882...(40 digits omitted)...9010', str_value='0x788c45dd60ae4dbe5055b5ac02384d5dc84677b2', hex_value='307837383863343564643630616534646265353035356235616330323338346435646338343637376232')}

In [8]: parser.get_literals('Delta', True).get('number')
Out[8]: {1, 10, 18, 1000}
```

Reply:
In the caller side do we lose the `literal.str_value` information? https://etherscan.io/address/0x000000000000ad05ccc4f10045630fb830b95127 address like this when defining will be ~42 char but convert to int literals it only has ~112 bits ~ 28 chars.

Reply:
ah ok, i will try that `parser.get_literals('Delta').get('number')`